### PR TITLE
Feature 169 settings polishing

### LINKED ISSUE #169: settings polishing
As a use I want the settings be polished

### PR DESCRIPTION
**Pull request by Chi-Chu319**
resolves #169

### CODE REVIEW

File: src/components/settings/settings.tsx [hunk 1]
```diff
@@ -222,17 +354,196 @@ class Settings extends React.Component<Props, State> {
...
-      conditions: conditions
+      userSettings: updatedUserSettings,
+      modified: false
     });
   };
 
+  /**
+   * On home address change handler
+   */
+  private onHomeAddressChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (Daniilmass) [inline src/components/settings/settings.tsx:430]: Add TSdocs "event" parameter

File: src/components/settings/settings.tsx [hunk 2]
```diff
@@ -222,17 +354,196 @@ class Settings extends React.Component<Props, State> {
...
+    this.setState({
+      userSettings: updatedUserSettings,
+      modified: true
+    });
+  };
+
+  /**
+   * On medical condition check handler
+   */
+  private onMedicalConditionCheck = (medicalConditionId: string) => (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (Daniilmass) [inline src/components/settings/settings.tsx:455]: Add TSdocs "medicalConditionId" parameter

**Review by Daniilmass — CHANGES_REQUESTED** [review]

File: src/components/settings/settings.tsx (hunk 1 above)
Comment (Chi-Chu319) [inline src/components/settings/settings.tsx:430]:
```suggestion
  /**
   * On home address change handler
   *
   * @param event event
   */
  private onHomeAddressChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

File: src/components/settings/settings.tsx (hunk 2 above)
Comment (Chi-Chu319) [inline src/components/settings/settings.tsx:455]:
```suggestion
  /**
   * On medical condition check handler
   *
   * @param medicalConditionId medical condition id
   * @param event event
   */
  private onMedicalConditionCheck = (medicalConditionId: string) => (event: React.ChangeEvent<HTMLInputElement>) => {
```

**Review by Daniilmass — APPROVED** [review]

**Pull request merged**